### PR TITLE
[ElmSharp] Add efl_util input generator APIs in ElmSharp

### PR DESCRIPTION
**Pull request by arosis78**
### Description of Change ###

Add new APIs for efl-util input generator APIs in ElmSharp

### Bugs Fixed ###

None

### API Changes ###

Added:
 - void InputGenerator.GenerateKeyEvent(string key, int pressed);
 - void InputGenerator.GenerateMouseEvent(int buttons, InputPointerType type, int x, int y)
 - void InputGenerator.GenerateTouchEvent(int index, InputTouchType type, int x, int y)

### Behavioral Changes ###

None

### CODE REVIEW

File: src/ElmSharp/ElmSharp/InputGenerator.cs [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+        /// </remarks>
+        All = (TouchScreen | Keyboard),
+
+        /// <summary>
+        /// Max.
+        /// </summary>
+        /// <remarks>
+        /// Deprecated since 3.0.
+        /// </remarks>
+        Max = (1 << 10),
```
Comment (myroot) [inline src/ElmSharp/ElmSharp/InputGenerator.cs]: I think, we don't need a Max value.

File: src/ElmSharp/ElmSharp/InputGenerator.cs [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+        /// </summary>
+        End,
+
+        /// <summary>
+        /// Max.
+        /// </summary>
+        /// <remarks>
+        /// Deprecated since 3.0.
+        /// </remarks>
+        Max = 10,
```
Comment (myroot) [inline src/ElmSharp/ElmSharp/InputGenerator.cs]: I think, we don't need a Max value.

File: src/ElmSharp/ElmSharp/InputGenerator.cs [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+        protected virtual void Dispose(bool disposing)
+        {
+            if (_isDisposed)
+                return;
+
+            if (disposing)
+            {
+                Interop.Eutil.efl_util_input_deinitialize_generator(_handle);
+            }
+
```
Comment (myroot) [inline src/ElmSharp/ElmSharp/InputGenerator.cs]: Usually unmanaged object always released ,whether disposing is true or not
Is there some issue when execute efl_util_input_deinitialize_generator API in GC thread?

File: src/ElmSharp/ElmSharp/InputGenerator.cs (hunk 1 above)
Comment (arosis78) [reply src/ElmSharp/ElmSharp/InputGenerator.cs]: Done

File: src/ElmSharp/ElmSharp/InputGenerator.cs (hunk 2 above)
Comment (arosis78) [reply src/ElmSharp/ElmSharp/InputGenerator.cs]: Done

File: src/ElmSharp/ElmSharp/InputGenerator.cs (hunk 3 above)
Comment (arosis78) [reply src/ElmSharp/ElmSharp/InputGenerator.cs]: Right. This API don't have thread safety, so there is issue when execute in GC thread.
But resource and device will be managed on window system when process is killed.

**Pull request merged**